### ISSUE TITLE
JMAP Support

### ISSUE THREAD

**Issue by Permanently** [NamelessMC/Nameless#3084]
### Describe the feature

[JMAP](https://jmap.io/) would be great to have for email protocol support. I'd try explain it, but I think the website will do a better job:

> JMAP is not a conversion of IMAP to JSON; it is a new protocol. It was designed to make much more efficient use of network resources, to be easier for developers to work with, and hopefully to make the best protocol for email an open standard once more. It's based on years of experience and real-world experimentation at Fastmail, and on talking to other major MUA/MTA developers to make sure we understand the common needs of the industry.

**Comment by Derkades** [comment]
Maybe some day into the future, but not now considering:
- There is no PHP client
- It doesn't have many advantages for NamelessMC specifically (e.g. IMAP being blocked by firewalls is only an issue on public WiFi, not in a datacenter)
- As far as I'm aware, no provider supports only JMAP but not IMAP
- No popular mail servers implement JMAP
- No popular desktop clients implement JMAP
- No popular mobile clients implement JMAP
